drivers/spi/spi_sedi.c

```diff
@@ -0,0 +1,415 @@
+		data_out = NULL;
+		data_in = (uint8_t *)ctx->rx_buf;
+		transfer_bytes = ctx->rx_len;
+		spi->tx_dummy_len -= transfer_bytes;
+	} else if (ctx->rx_len == 0) {
+		/* tx only, nothing to rx */
+		data_out = (uint8_t *)ctx->tx_buf;
+		data_in = NULL;
+		transfer_bytes = ctx->tx_len;
+		spi->rx_dummy_len -= transfer_bytes;
+	} else if (ctx->tx_len == ctx->rx_len) {
+		/* rx and tx are the same length */
+		data_out = (uint8_t *)ctx->tx_buf;
+		data_in = (uint8_t *)ctx->rx_buf;
+		transfer_bytes = ctx->tx_len;
+	} else if (ctx->tx_len > ctx->rx_len) {
+		/* Break up the tx into multiple transfers so we don't have to
+		 * rx into a longer intermediate buffer. Leave chip select
+		 * active between transfers.
+		 */
+		data_out = (uint8_t *)ctx->tx_buf;
+		data_in = ctx->rx_buf;
+		transfer_bytes = ctx->rx_len;
+	} else {
+		/* Break up the rx into multiple transfers so we don't have to
+		 * tx from a longer intermediate buffer. Leave chip select
+		 * active between transfers.
+		 */
+		data_out = (uint8_t *)ctx->tx_buf;
+		data_in = ctx->rx_buf;
+		transfer_bytes = ctx->tx_len;
+	}
+
+	spi_context_cs_control(&spi->ctx, false);
+
+	ret = sedi_spi_transfer(info->spi_device, data_out, data_in,
+				transfer_bytes);
+
+	if (ret != SEDI_DRIVER_OK) {
+		goto out;
+	}
+
+	ret = spi_context_wait_for_completion(&spi->ctx);
+	if (ret != 0) {
+		sedi_spi_status_t spi_status = {0};
+
+		sedi_spi_get_status(info->spi_device, &spi_status);
+
+		/* SPI ABORT */
+		sedi_spi_control(info->spi_device, SEDI_SPI_IOCTL_ABORT, 0);
+		/* Toggle GPIO back */
+		spi_context_cs_control(&spi->ctx, true);
+	}
+out:
+	spi_context_release(&spi->ctx, ret);
+	pm_device_busy_clear(dev);
+
+	return ret;
+}
+
+static int spi_sedi_transceive(const struct device *dev,
+			       const struct spi_config *config,
+			       const struct spi_buf_set *tx_bufs,
+			       const struct spi_buf_set *rx_bufs)
+{
+	return transceive(dev, config, tx_bufs, rx_bufs, false, NULL, NULL);
+}
+
+#ifdef CONFIG_SPI_ASYNC
+static int spi_sedi_transceive_async(const struct device *dev,
+				     const struct spi_config *config,
+				     const struct spi_buf_set *tx_bufs,
+				     const struct spi_buf_set *rx_bufs,
+				     spi_callback_t cb,
+				     void *userdata)
+{
+	return transceive(dev, config, tx_bufs, rx_bufs, true, cb, userdata);
+}
+#endif /* CONFIG_SPI_ASYNC */
+
+static int spi_sedi_release(const struct device *dev,
+			    const struct spi_config *config)
+{
+	struct spi_sedi_data *spi = dev->data;
+
+	if (!spi_context_configured(&spi->ctx, config)) {
+		return -EINVAL;
+	}
+
+	spi_context_unlock_unconditionally(&spi->ctx);
+
+	return 0;
+}
+
+extern void spi_isr(sedi_spi_t device);
+
+void spi_sedi_callback(uint32_t event, void *param)
+{
+	const struct device *dev = (const struct device *)param;
+	const struct spi_sedi_config *info = dev->config;
+	struct spi_sedi_data *spi = dev->data;
+	struct spi_context *ctx = &spi->ctx;
+	int error;
+
+	if (event == SEDI_SPI_EVENT_DATA_LOST) {
```

Comment:
Seems like this branching over event could easily be done using a switch, even for the error case, not really a big deal the way its written but might be a little nicer

Reply:
Yes Thanks Teburd for the comment. A switch could be a little better, but it won't make much difference here because there are only 2 different cases. I will choose not to push another patch to replace it this time.